facebook/delphiFacebook/R/contingency_variables.R

```diff
@@ -406,7 +504,402 @@ create_derivative_columns <- function(df) {
+    df$b_hesitant_barrier_distrust_vaccines <- as.numeric(is_selected(hesitancy_reasons, "10"))
+    df$b_hesitant_barrier_distrust_govt <- as.numeric(is_selected(hesitancy_reasons, "11"))
+    df$b_hesitant_barrier_health_condition <- as.numeric(is_selected(hesitancy_reasons, "12"))
+    df$b_hesitant_barrier_other <- as.numeric(is_selected(hesitancy_reasons, "13"))
+    df$b_hesitant_barrier_pregnant <- as.numeric(is_selected(hesitancy_reasons, "14"))
+    df$b_hesitant_barrier_religious <- as.numeric(is_selected(hesitancy_reasons, "15"))
+  } else {
+    df$b_hesitant_barrier_sideeffects <- NA_real_
+    df$b_hesitant_barrier_allergic <- NA_real_
+    df$b_hesitant_barrier_ineffective <- NA_real_
+    df$b_hesitant_barrier_dontneed <- NA_real_
+    df$b_hesitant_barrier_dislike_vaccines <- NA_real_
+    df$b_hesitant_barrier_not_recommended <- NA_real_
+    df$b_hesitant_barrier_wait_safety <- NA_real_
+    df$b_hesitant_barrier_low_priority <- NA_real_
+    df$b_hesitant_barrier_cost <- NA_real_
+    df$b_hesitant_barrier_distrust_vaccines <- NA_real_
+    df$b_hesitant_barrier_distrust_govt <- NA_real_
+    df$b_hesitant_barrier_health_condition <- NA_real_
+    df$b_hesitant_barrier_other <- NA_real_
+    df$b_hesitant_barrier_pregnant <- NA_real_
+    df$b_hesitant_barrier_religious <- NA_real_
+  }
+  
+  # selected "I don't believe I need a COVID-19 vaccine." in any of V5b, V5c
+  # AND reasons why you don't believe you need a COVID-19 vaccine
+  # V5b, V5c, V6
+  if (all(c("V5b", "V5c", "V6", "b_hesitant_barrier_dontneed") %in% names(df))) {
+    dontneed <- df$b_hesitant_barrier_dontneed == 1
+    dontneed_reasons <- split_options(df$V6)
+    
+    df$b_hesitant_dontneed_reason_had_covid <- as.numeric(dontneed & is_selected(dontneed_reasons, "1"))
+    df$b_hesitant_dontneed_reason_dont_spend_time <- as.numeric(dontneed & is_selected(dontneed_reasons, "2"))
+    df$b_hesitant_dontneed_reason_not_high_risk <- as.numeric(dontneed & is_selected(dontneed_reasons, "3"))
+    df$b_hesitant_dontneed_reason_precautions <- as.numeric(dontneed & is_selected(dontneed_reasons, "4"))
+    df$b_hesitant_dontneed_reason_not_serious <- as.numeric(dontneed & is_selected(dontneed_reasons, "5"))
+    df$b_hesitant_dontneed_reason_not_beneficial <- as.numeric(dontneed & is_selected(dontneed_reasons, "7"))
+    df$b_hesitant_dontneed_reason_other <- as.numeric(dontneed & is_selected(dontneed_reasons, "8"))
+  } else {
+    df$b_hesitant_dontneed_reason_had_covid <- NA_real_
+    df$b_hesitant_dontneed_reason_dont_spend_time <- NA_real_
+    df$b_hesitant_dontneed_reason_not_high_risk <- NA_real_
+    df$b_hesitant_dontneed_reason_precautions <- NA_real_
+    df$b_hesitant_dontneed_reason_not_serious <- NA_real_
+    df$b_hesitant_dontneed_reason_not_beneficial <- NA_real_
+    df$b_hesitant_dontneed_reason_other <- NA_real_
+  }
+  
+  # C14
+  if ("b_public_mask_often" %in% names(df)) {
+    df$b_wearing_mask <- as.numeric(df$b_public_mask_often == 1)
```

Comment:
Need a `_7d` version of this variable to reflect a change in the question wording.

Reply:
I believe there's already a `b_public_mask_often_7d` (or similar) variable defined in the `load_responses` process, FYI.

Reply:
Added `wearing_mask_5d` and `wearing_mask_7d`.